client/extensions/woocommerce/lib/generate-variations/test/index.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import generateVariations from '../index';
+
+describe( 'generateVariations', () => {
```

Comment:
@coderkevin https://github.com/Automattic/wp-calypso/pull/13436/files#diff-c35d147d05f7b3cae0bacb4753172e53R96 does that. It has three types: Color, Size, Cut and generates 4 combos:

Red - Small - Unisex
Red - Medium - Unisex
Blue - Small - Unisex
Blue - Medium - Unisex

Do you think I need to expand this at all?